extension/storage/filestorage/client.go

```diff
@@ -63,6 +65,9 @@ func newClient(logger *zap.Logger, filePath string, timeout time.Duration, compa
 	}
 
 	client := &fileStorageClient{logger: logger, db: db, compactionCfg: compactionCfg, openTimeout: timeout}
+	if compactionCfg.CleanupOnStart {
+		client.cleanup(compactionCfg.Directory)
+	}
```

Comment:
One more thing - we should add a `Start` function so that cleanup is actually run on start, rather than on creation of the component.

Here's an example: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c6a6bd43875769d7400e8462e63730fe1a49ab11/extension/pprofextension/pprofextension.go#L31

Reply:
Thanks for your patience, that hint was eye opening and could understand better the nuances of this extension lifecycle. 
It also address an issue according running the test on linux that was no simulating locked files. 
Please, take another look.

Reply:
Thanks for iterating on it.